proposals/p5270.md

```diff
@@ -0,0 +1,132 @@
+
+The explorer codebase is no longer actively developed or maintained, but retains
+value as a place to look for implementations of various parts of the design.
+Having both the explorer and toolchain in the working git repository causes
+confusion for developers and users, and increases our incremental maintenance
+burden. We will move explorer out of the working git repository so that it
+remains accessible but is clearly deliniated from the toolchain and is more
+clearly frozen or archived.
+
+## Problem
+
+Users have expressed confusion when they come across examples or tests for the
+explorer that involve `Main()`, whereas the design and implementation of the
+toolchain have moved on to `Run()` as the entry point.
+
+Developers have attempted to update the explorer codebase to fix style guide
+issues[^1], as tools like grep or clang-tidy do not differentiate our active
+toolchain codebase from the frozen explorer one.
+
+As we update Bazel, Clang, Clang-tidy, etc, we also have to update the explorer
+codebase to keep it building cleanly.
+
+As we modify the `//testing` harness, we have to accommodate and work around the
+explorer tests[^2][^3][^4].
+
+[^1]: https://github.com/carbon-language/carbon-lang/pull/5224
+
+[^2]: https://github.com/carbon-language/carbon-lang/pull/4979
+
+[^3]: https://github.com/carbon-language/carbon-lang/pull/5025
+
+[^4]: https://github.com/carbon-language/carbon-lang/pull/5036
+
+## Background
+
+-   Previous proposal:
+    [p3532: Focus implementation effort on the toolchain](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p3532.md)
+-   Kick-off of this discussion
+    [#5224](https://github.com/carbon-language/carbon-lang/pull/5224#pullrequestreview-2730512195)
+-   Discord discussion:
+    [2025-04-02 #explorer](https://discord.com/channels/655572317891461132/763516049710120960/1356979197070803095)
+
+## Proposal
+
+1. Add a tag `explorer-archived` in the main `carbon-lang` git repository.
+2. Copy the `//explorer` directory and its dependencies to a new `explorer`
+   repository under the `carbon-language` organization.
+3. Locally ensure the explorer tests build and pass under the `explorer`
+   repository.
+4. Add a `README.md` to the `explorer` repository that explains explorer is
+   archived and not under active development.
+5. Delete `//explorer` in the main `carbon-lang` repository.
+
+Note that fuzzer test cases from the explorer are already relocated under
+`//toolchain/*/fuzzer_corpus/`.
+
+## Rationale
+
+The primary purpose of the explorer codebase at this time is as a demonstration
+of past work implementing the carbon language design. This can help to inform
+the implementation of the toolchain as it catches up in various areas, as long
+as the design has not deviated from the explorer implementation.
+
+Searching in, and providing links to the explorer codebase comes up in design
+discussions occasionally [^5], and we should maintain the ability of toolchain
```

Comment:
Note, I really wish we could come up with more than one example use over the past year if we're going to maintain an archive repo. Given the reference's use as GH search rationale, maybe @zygoloid can confirm that he used GH search to find the reference, and that maintaining search is worth a small cost?

Reply:
Ok, thanks, I found a bunch more, even [one from](https://discord.com/channels/655572317891461132/998959756045713438/1225116234199203860) @geoffromer.